lib/types/library/LibraryFormatter.js

```diff
@@ -1,75 +1,178 @@
+	 * @throws {Error} if namespace can not be determined
+	 */
+	async getNamespace() {
+		let appId;
+		try {
+			const manifest = await this.getManifest();
+			// check for a proper sap.app/id in manifest.json to determine namespace
+			if (manifest["sap.app"] && manifest["sap.app"].id) {
+				appId = manifest["sap.app"] && manifest["sap.app"].id;
 			} else {
-				log.verbose(`Ignoring 'test' directory for project ${project.metadata.name}.` +
-					"Either no setting was provided or the path not found.");
-			}
-		}).then(() => {
-			if (project.metadata.copyright) {
-				return;
+				log.verbose(
+					`No "sap.app" ID configuration found in manifest.json of project ${this._project.metadata.name}`);
 			}
-			// If no copyright replacment was provided by ui5.yaml,
-			// check if the .library file has a valid copyright replacement
-			const basePath = path.join(project.path, project.resources.pathMappings["/resources/"]);
-			return glob("**/.library", {
-				cwd: basePath
-			}).then((dotLibraryResources) => {
-				if (!dotLibraryResources.length) {
-					log.verbose(`Could not find .library file for project ${project.id}`);
-					return;
-				}
-				if (dotLibraryResources.length > 1) {
-					throw new Error(`Found multiple (${dotLibraryResources.length}) .library files ` +
-						`for project ${project.id}`);
+		} catch (err) {
+			log.verbose(`Namespace resolution from manifest.json failed for project ` +
+				`${this._project.metadata.name}: ${err.message}`);
+			log.verbose(`Falling back to .library file...`);
+		}
+
+		if (!appId) {
+			try {
+				const dotLibrary = await this.getDotLibrary();
+				if (dotLibrary && dotLibrary.library && dotLibrary.library.name) {
+					appId = dotLibrary.library.name;
+				} else {
+					throw new Error(
+						`No library name found in .library of project ${this._project.metadata.name}`);
 				}
-				const fsPath = path.join(basePath, dotLibraryResources[0]);
-				return readFile(fsPath)
-					.then((content) => xmlParse(content))
-					.then((result) => {
-						if (result && result.library.copyright) {
-							project.metadata.copyright = result.library.copyright[0];
-							log.verbose(`Use copyright from .library for project ${project.id}`);
-						}
-					}, (error) => {
-						throw new Error(`Failed to check .library file for project ${project.id}, ` +
-							`error: ${error.message}`);
-					});
+			} catch (err) {
+				throw new Error(`Namespace resolution from .library failed for project ` +
+					`${this._project.metadata.name}: ${err.message}`);
+			}
+		}
+
+		if (this.hasMavenPlaceholder(appId)) {
+			try {
+				appId = await this.resolveMavenPlaceholder(appId);
+			} catch (err) {
+				throw new Error(
+					`Failed to resolve namespace maven placeholder of project ` +
+					`${this._project.metadata.name}: ${err.message}`);
+			}
+		}
+
+		const namespace = appId.replace(/\./g, "/");
+		log.verbose(`Namespace of project ${this._project.metadata.name} is ${namespace}`);
```

Comment:
Also check whether the library namespace matches with the file path of `manifest.json`, `.library` or `library.js`

Reply:
Done